client/signup/steps/plans/index.jsx

```diff
@@ -188,7 +189,31 @@ export class PlansStep extends Component {
 	}
 
 	getSubHeaderText() {
-		const { translate, useEmailOnboardingSubheader } = this.props;
+		const { translate, useEmailOnboardingSubheader, signupDependencies, flowName } = this.props;
+
+		const { segmentationSurveyAnswers } = signupDependencies;
+
+		const surveyedIntent =
+			flowName === 'guided' ? getSegmentedIntent( segmentationSurveyAnswers ) : undefined;
+
+		if ( surveyedIntent === 'plans-guided-segment-developer-or-agency' ) {
+			const a4aLinkButton = (
+				<Button
+					href={ localizeUrl( 'https://automattic.com/for-agencies/' ) }
+					target="_blank"
+					rel="noopener noreferrer"
+					onClick={ () =>
+						this.props.recordTracksEvent( 'calypso_guided_onboarding_agency_link_click' )
+					}
+					borderless
+				/>
+			);
+
+			return translate(
+				'Are you an agency? Get the best of WordPress with {{link}}Automattic for Agencies{{/link}}',
+				{ components: { link: a4aLinkButton } }
+			);
+		}
```

Comment:
To make this easier to remove later, can we throw this whole chunk in [`util.ts`](https://github.com/Automattic/wp-calypso/blob/add/trailmap-expirement/client/signup/steps/plans/util.ts#L28)?